crates/context_aware_config/src/api/default_config/types.rs

```diff
@@ -10,6 +12,38 @@ pub struct CreateReq {
     pub function_name: Option<Value>,
 }
 
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "String")]
+pub struct DefaultConfigKey(String);
+impl DefaultConfigKey {
+    pub fn validate_data(name: String) -> Result<Self, String> {
+        let regex_str: &str = "^[a-zA-Z0-9-_]([a-zA-Z0-9-_.]{0,254}[a-zA-Z0-9-_])?$";
+
+        let regex = Regex::new(regex_str).map_err(|err| {
+            log::error!("error while validating defaultConfigKey : {err}");
+            "Something went wrong".to_string()
+        })?;
+
+        if !regex.is_match(&name) {
+            return Err(format!(
+                "The key name {} is invalid, it should obey the regex {}. \
```

Comment:
Not mandatory, but should we start considering this syntax for simple variable values ?
```suggestion
                "The key name {name} is invalid, it should obey the regex {regex_str}. \
```